test/binaryen.js/exception-handling.js

```diff
@@ -18,7 +18,8 @@ function stringify(expr) {
 
 function test() {
   var module = new Binaryen.Module();
-  module.setFeatures(Binaryen.Features.ExceptionHandling);
+  module.setFeatures(Binaryen.Features.ReferenceTypes |
+                     Binaryen.Features.ExceptionHandling);
```

Comment:
i wonder if enabling exceptions should automatically enable reference types? not sure what the plan in this area is...

Reply:
Exception handling is dependent on reference types, but we currently have separate flags, like other tools: wabt or v8. I'd like to keep those feature flags separate and not dependent on each other for consistency, and the recommended usage is to control the whole thing using clang's (and emscripten's) `-fwasm-exception` flag, which will [turn on both exception handling and reference type features](https://github.com/llvm/llvm-project/blob/8304781cae97f72c737610460bcb25aa2082807d/clang/lib/Driver/ToolChains/WebAssembly.cpp#L240-L244) and embed the info into the custom section. But here in the test there's no way to turn on both features, so..